apps/web/src/application/services/SwitchNetwork/SwitchNetworkService.ts

```diff
@@ -0,0 +1,50 @@
+import { NetworkEnum } from '@cryptochords/shared'
+import { Network } from '../../../domain/entities/Network'
+import { CubeRepository } from '../../../domain/repositories/CubeRepository'
+import { NetworkRepository } from '../../../domain/repositories/NetworkRepository'
+import { TransactionRepository } from '../../../domain/repositories/TransactionRepository'
+import { ObservableService } from '../../ObservableService'
+import { SwitchNetworkRequestDto } from './SwitchNetworkRequestDto'
+import { SwitchNetworkResponseDto } from './SwitchNetworkResponseDto'
+
+export class SwitchNetworkService extends ObservableService<SwitchNetworkRequestDto, SwitchNetworkResponseDto> {
+  private readonly transactionRepository: TransactionRepository
+  private readonly networkRepository: NetworkRepository
+  private readonly cubeRepository: CubeRepository
+
+  constructor(
+    transactionRepository: TransactionRepository,
+    networkRepository: NetworkRepository,
+    cubeRepository: CubeRepository
+  ) {
+    super()
+    this.transactionRepository = transactionRepository
+    this.networkRepository = networkRepository
+    this.cubeRepository = cubeRepository
+  }
+
+  protected async process(request: SwitchNetworkRequestDto): Promise<SwitchNetworkResponseDto> {
+    const network = await this.validateNetwork(request.networkName)
+    await this.transactionRepository.clear()
+    await this.cubeRepository.clear()
+    await this.networkRepository.select(network.name)
```

Comment:
Can these 3 be run in parallel? As they include side effects (at least it looks like that given their names), they may not, but maybe they can 😅

Reply:
Yes they can run in parallel. Thanks!